3rdparty/osqp/CMakeLists.txt

```
@@ -0,0 +1,14 @@
+cmake_minimum_required(VERSION 2.8.3)
+project(osqp)
+
+include(ExternalProject)
+
+ExternalProject_Add(
+  osqp
+  URL https://github.com/osqp/osqp/releases/download/v0.6.2/complete_sources.tar.gz
+  URL_HASH MD5=66738e622174eb506e0cee106891397e
+  CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
+  INSTALL_COMMAND echo "install"
+  )
+
+install(CODE "execute_process(COMMAND make install WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/osqp-prefix/src/osqp-build)")
```

2. skip build indigo in CMakelists
```suggestion
if("$ENV{ROS_DISTRO}" STRGREATER "kinetic")
  ExternalProject_Add(
    osqp
    URL https://github.com/osqp/osqp/releases/download/v0.6.2/complete_sources.tar.gz
    URL_HASH MD5=66738e622174eb506e0cee106891397e
    CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
    INSTALL_COMMAND echo "install"
    )

  install(CODE "execute_process(COMMAND make install WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/osqp-prefix/src/osqp-build)")
else()
  install(CODE "message(WARNING \"Skipping osqp install because ROS is too old\")")
endif()
```

---

Maybe we can try patch first.

---

もしも @tongtybj 先生がpatchを作るつもりだったのでしたら僕が勝手なことをしてしまって申し訳ないです. https://github.com/jsk-ros-pkg/jsk_3rdparty/pull/442/commits/d92b5f31db4e33f4b8d3affcc48d9ced004a5810 は消してください。

- indigoでosqpを使いたい人がいる
- osqpをindigoでビルドできるようにする作業をしたい人がいる

のどちらかで無ければ、osqpをjsk_3rdpartyに移す目的の達成のためには2. skip build indigo in CMakelists で必要十分と思います。